vault/expiration.go

```diff
@@ -102,6 +102,7 @@ func expireLeaseStrategyRevoke(ctx context.Context, m *ExpirationManager, le *le
 			case <-ctx.Done():
 			case <-m.quitCh:
 				cancel()
+			case <-revokeCtx.Done():
```

Comment:
This seems right but should probably then also put `cancel()` calls into each of the select cases below.